src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -0,0 +1,435 @@
+        reinterpret_cast<const unsigned char*>(ui_data.data());
+    bulletMesh.m_triangleIndexStride = 3 * sizeof(Magnum::UnsignedInt);
+    bulletMesh.m_numVertices = v_data.size();
+    //! Get the pointer to the first float of the first triangle
+    bulletMesh.m_vertexBase =
+        reinterpret_cast<const unsigned char*>(v_data.data()->data());
+    bulletMesh.m_vertexStride = sizeof(Magnum::Vector3);
+    bulletMesh.m_indexType = PHY_INTEGER;
+    bulletMesh.m_vertexType = PHY_FLOAT;
+
+    btTransform t;  // position and rotation
+    t.setIdentity();
+    t.setOrigin(btVector3(0, 0, 0));
+    //! Create convex component
+    bObjectConvexShapes_.emplace_back(std::make_unique<btConvexHullShape>(
+        static_cast<const btScalar*>(meshData.positions.data()->data()),
+        meshData.positions.size(), sizeof(Magnum::Vector3)));
+    bObjectConvexShapes_.back()->setMargin(margin);
+    //! Add to compound shape stucture
+    bObjectShape_->addChildShape(t, bObjectConvexShapes_.back().get());
+  }
+  //! Set properties
+  bObjectShape_->setMargin(margin);
+
+  btVector3 bInertia =
+      btVector3(physicsObjectAttributes.getMagnumVec3("inertia"));
+
+  if (bInertia[0] == 0. && bInertia[1] == 0. && bInertia[2] == 0.) {
+    // Alex TODO: allow bullet to compute the inertia tensor if we don't have
+    // one
+    bObjectShape_->calculateLocalInertia(
+        physicsObjectAttributes.getDouble("mass"),
+        bInertia);  // overrides bInertia
+    LOG(INFO) << "Automatic object inertia computed: " << bInertia.x() << " "
+              << bInertia.y() << " " << bInertia.z();
+  } else {
+    LOG(INFO) << "User provided object inertia " << bInertia.x() << " "
+              << bInertia.y() << " " << bInertia.z();
+  }
+
+  //! Bullet rigid body setup
+  bObjectMotionState_ = new Magnum::BulletIntegration::MotionState(*this);
+  btRigidBody::btRigidBodyConstructionInfo info =
+      btRigidBody::btRigidBodyConstructionInfo(
+          physicsObjectAttributes.getDouble("mass"),
+          &(bObjectMotionState_->btMotionState()), bObjectShape_.get(),
+          bInertia);
+  info.m_friction = physicsObjectAttributes.getDouble("frictionCoefficient");
+  info.m_restitution =
+      physicsObjectAttributes.getDouble("restitutionCoefficient");
+  info.m_linearDamping = physicsObjectAttributes.getDouble("linDamping");
+  info.m_angularDamping = physicsObjectAttributes.getDouble("angDamping");
+  // Magnum::Vector3 inertia = metaData.inertia;
+  // info.m_localInertia   = bInertia(inertia.x(), inertia.y(), inertia.z());
+
+  //! Create rigid body
+  bObjectRigidBody_ = std::make_unique<btRigidBody>(info);
+  //! Add to world
+  bWorld->addRigidBody(bObjectRigidBody_.get());
+  //! Sync render pose with physics
+  bWorld_ = bWorld;
+  initialized_ = true;
+  syncPose();
+  return true;
+}  // end BulletRigidObject::initializeObject
+
+bool BulletRigidObject::removeObject() {
+  bWorld_->removeRigidBody(bObjectRigidBody_.get());
+  bWorld_ = nullptr;
```

Comment:
Curious -- why? :) Both operations are equivalent and I like to use the `= nullptr` (or even `= {}`) because (for me at least) it doesn't make it any less clear.